rust/src/smb/smb.rs

```diff
@@ -947,6 +951,8 @@ impl SMBState {
                 false
             };
             if found {
+                tx.tx_data.updated_tc = true;
```

Comment:
why do we set both sides to true here? I guess this is a generic question, as many of the patterns appear to always set both to true

Reply:
For this specific case : The name `get_generic_tx` did not indicate a direction. Apparently, all the usages are updating with responses... Should I change only `updated_tc` and change the name of the function, so that it does not get misused ?

For the generic question, the answer is that it is correct to do both sides even if it is optimal to do only one side.
This whole PR is already an optimization with some risk of overoptimizing and being incorrect, so I stayed on the safe side in the cases where the direction was not obvious like here...
What do you think ?